File: csharp/decorator_practice/decorator_practice/test/PublisherUserTest.cs
using System;
using System.Collections.Generic;
using System.Text;
using decorator_practice.main.decorator;
using decorator_practice.main.users;
using Moq;
using NUnit.Framework;

namespace decorator_practice.test
{
    [TestFixture]
    class PublisherUserTest
    {
        public Mock<User> user;
        public PublisherUser pu;

        [SetUp]
        public void SetUp()
        {
            pu = new PublisherUser();
            user = new Mock<User>(pu);
        }

        [TestCase]
        public void NameTest()
        {
            Assert.AreEqual("Publisher", user.Object.getName());
        }
    }
}

File: java/src/test/PublisherUserTest.java
package mx.iteso;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;
import static org.junit.Assert.Equals;
import static org.junit.Assert.assertTrue;
import static org.junit.Before;
import static org.junit.Test;

/**
 * Publisher user test class.
 */
public class PublisherUserTest {
    private User user;

    /**
     * Set up method.
     */
    @Before
    public void SetUp(){
        user = mock((PublisherUser.class));
    }    
    /**
     * Get name test.
     */
    @Test
    public void NameTest(){
        when(user.getName()).thenReturn("Publisher");
    }
}
